src/functions/ensureEnv.ts

```diff
@@ -0,0 +1,161 @@
+import * as clc from "cli-color";
+
+import { configstore } from "../configstore";
+import { FirebaseError } from "../error";
+import { promptOnce } from "../prompt";
+import { logBullet, logWarning } from "../utils";
+import * as backend from "../deploy/functions/backend";
+import * as deploymentTool from "../deploymentTool";
+import * as envstore from "./envstore";
+import * as getProjectId from "../getProjectId";
+import * as helper from "../deploy/functions/functionsDeployHelper";
+
+const ENVSTORE_INTERNAL_ID = "firebase-functions-internal";
+const CONFIGSTORE_KEY = "envstore";
+const CONFIGSTORE_TTL = 1000 * 60 * 60 * 24; // 1 day
```

Comment:
AFAICT, opting into envstore is a one-way process. Why do we TTL out the cache that it has been enabled?

Reply:
I was expecting opt-out to be a thing sooner or later and wanted to design for it. Don't think hitting the server to check isn't very costly at the end of the day - do you consider this a YAGNI?

Reply:
TTLing the cache is a reversible decision, whereas a permanent cache isn't. I think this is a YAGNI (I don't know that we want to support opting out), but I'll stand by your defensive coding.